v03_pipeline/lib/misc/family_entries.py

```diff
@@ -0,0 +1,154 @@
+import hail as hl
+
+from v03_pipeline.lib.model import DatasetType
+
+
+def compute_callset_family_entries_ht(
+    dataset_type: DatasetType,
+    mt: hl.MatrixTable,
+    entries_fields: dict[str, hl.Expression],
+) -> hl.Table:
+    sample_id_to_family_guid = hl.dict(
+        {
+            s: family_guid
+            for family_guid, sample_ids in hl.eval(mt.family_samples).items()
+            for s in sample_ids
+        },
+    )
+    ht = mt.select_rows(
+        filters=mt.filters.difference(dataset_type.excluded_filters),
+        family_entries=(
+            # NB: we're sorted by both family and sample when this runs.
+            # However, the sort is not guaranteed once the entries
+            # table is editted and families are spliced out and re-appended.
```

Comment:
if the sort is not maintained in the long run, is there an advantage to doing it at all, or is it just a slight performance drain?

Reply:
I'm not convinced we need it at all, but it does add some determinism to the method (since we group by family_guid into a dictionary and then flatten the values).  

It's not (or at least shouldn't be) a performance drain.